### PR TITLE
docs: add example of Azure deployment with Static Web Apps

### PR DESCRIPTION
**Pull request by kark**
#### Summary

PR adds example of Azure deployment in the docs. 
Closes [#2683](https://github.com/commercetools/merchant-center-application-kit/issues/2683)

### CODE REVIEW

File: website/src/content/deployment-examples/azure.mdx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  "env": {
+    "production": {
+      "applicationId": "${env:APPLICATION_ID}",
+      "url": "https://<project>.1.azurestaticapps.net"
+    }
+  }
+}
+```
+
+Create the environment variables in the Microsoft Azure application settings: **Home > Static Web Apps > Your project name > Configuration > Application Settings**.
```
Comment (CarlosCortizasCT) [inline website/src/content/deployment-examples/azure.mdx:48]: Here we are mentioning the `Static Web Apps` before mentioning (in the next section) that this is our preferred way of integration in Azure.
Not sure if it we would need to move this part to the next section.

**Review by CarlosCortizasCT — COMMENTED** [review]
This looks very good 👏 

I just left some suggestions to maybe take into consideration.

File: website/src/content/deployment-examples/azure.mdx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+---
+title: Deploying to Microsoft Azure
+---
+
+This deployment example refers to [Microsoft Azure](https://azure.microsoft.com/).
+
+# Prerequisites
```
[the remark this answers is not in the document]
Comment (emmenko) [reply website/src/content/deployment-examples/azure.mdx:7]: We didn't mention this in any of the other examples (also with having a github account), as it's not really a requirement and usually it's documented in the service itself when we refer to the GitHub integration.

If we decide to mention it, we should do it for every example then. Personally I think we can omit it as it's kind of implied in a sense.

File: website/src/content/deployment-examples/azure.mdx (hunk 1 above)
Comment (emmenko) [reply website/src/content/deployment-examples/azure.mdx:48]: Good point. This is actually the same structure as in some of the other examples.

If we want to change it, please remember to also adjust the other related documents.

**Review by emmenko — COMMENTED** [review]
Thanks for looking into that!

Overall looks very good, just some minor points to discuss as Carlos suggested.

File: website/src/content/deployment-examples/azure.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kark) [reply website/src/content/deployment-examples/azure.mdx:7]: Sure, got you, I removed that part 👍 I added it before only because it's explicitly mentioned in the [Azure's docs](https://docs.microsoft.com/en-us/azure/static-web-apps/get-started-portal?tabs=vanilla-javascript&pivots=github#prerequisites), and their infrastructure provisioning seems to be only working with GitHub

File: website/src/content/deployment-examples/azure.mdx (hunk 1 above)
Comment (kark) [reply website/src/content/deployment-examples/azure.mdx]: Sure, this makes sense. I updated the structure. Please let me know if you still consider that out of place after the changes

**Review by emmenko — COMMENTED** [review]
Thanks for the updates. Just a small remark otherwise the rest looks good 👌

File: website/src/content/deployment-examples/netlify.mdx [hunk 3]
```diff
@@ -29,23 +29,6 @@ Moreover, you need to provide the [production URL](/api-reference/application-co
...
-  "env": {
-    "production": {
-      "applicationId": "${env:APPLICATION_ID}",
-      "url": "https://<site>.netlify.app"
-    }
-  }
-}
-```
-
-Make sure then to provide the environment variables in your **Netlify site > Settings > Build & deploy > Environment**.
```
[the remark this answers is not in the document]
Comment (kark) [reply website/src/content/deployment-examples/netlify.mdx:47]: Updated in 5385e9fe225a44fc8f7e0bdafd70197365a170b2

File: website/src/content/deployment-examples/cloudflare.mdx [hunk 4]
```diff
@@ -82,6 +80,12 @@ In the Cloudflare Pages setup process you need to configure the following things
 * Override the *Output directory* with: `public`.
 * If possible, select or specify the *Node.js Version*. Recommended version is `>= v14`.
 
+<Info>
+
+If your Custom Application config [requires environment variables](#using-environment-variables), make sure to provide them in your **Account Home > Pages > Pages project > Settings > Environment variables**.
```
Comment (emmenko) [inline website/src/content/deployment-examples/cloudflare.mdx:85]: @kark FYI: I rephrased the sentence a bit to make it more clear.

**Review by emmenko — COMMENTED** [review]
I think this is good to go now. One last question maybe to clarify.

File: website/src/content/deployment-examples/azure.mdx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  "env": {
+    "production": {
+      "applicationId": "${env:APPLICATION_ID}",
+      "url": "https://<project>.1.azurestaticapps.net"
+    }
+  }
+}
+```
+
+# Connect Microsoft Azure with GitHub
```
[the remark this answers is not in the document]
Comment (kark) [reply website/src/content/deployment-examples/azure.mdx:47]: Thanks, that is true. There are 3 ways of deploying Azure Static Web Apps:
- using [Visual Studio Code extension](https://docs.microsoft.com/en-us/azure/static-web-apps/getting-started?tabs=react)
- using Azure portal:
  - [GitHub repo](https://docs.microsoft.com/en-us/azure/static-web-apps/get-started-portal?tabs=react&pivots=github) 👈 **this subversion is covered**
  - [Azure DevOps](https://docs.microsoft.com/en-us/azure/static-web-apps/get-started-portal?tabs=react&pivots=azure-devops)
- [using Azure CLI](https://docs.microsoft.com/en-us/azure/static-web-apps/get-started-cli?tabs=react)

So would you prefer the title that you suggested `Deploying to Azure with Static Web Apps` or perhaps would you like it to be even more specific `Deploying to Azure with Static Web Apps from GitHub repository`?

BTW. I might cover other options as well (if needed).

[the remark this answers is not in the document]
Comment (emmenko) [reply website/src/content/deployment-examples/azure.mdx:47]: Ok, I think we can then rename the page to `Deploying to Azure with Static Web Apps` and have more subsections related to the different deployment options (as a follow up).

So the section `Connect Microsoft Azure with GitHub` could become `Connect Azure with GitHub Actions` and have others such as `Connect Azure with Azure DevOps`, etc.

What do you think?

[the remark this answers is not in the document]
Comment (kark) [reply website/src/content/deployment-examples/azure.mdx]: Makes sense 👍 Updated in a846d3b5c91b3231d29933b94549a083eea29ec3

File: website/src/content/deployment-examples/azure.mdx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+Before you get started, you need to have:
+
+- A [Microsoft Azure](https://azure.microsoft.com/free/) account.
+- A [commercetools](https://docs.commercetools.com/getting-started) account and a Project.
+- A Custom Application [configured in the Merchant Center](https://docs.commercetools.com/merchant-center/managing-custom-applications).
+
+# Configuration
+
+In your Custom Application config, make sure to provide the [Custom Application ID](/api-reference/application-config#envproductionapplicationid) that you got when you configured the Custom Application in the Merchant Center.
```
Comment (Rhotimee) [inline website/src/content/deployment-examples/azure.mdx:17]:
```suggestion
In your Custom Application config, provide the [Custom Application ID](/api-reference/application-config#envproductionapplicationid) you got when you configured the Custom Application in the Merchant Center.
```

File: website/src/content/deployment-examples/azure.mdx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+Moreover, you need to provide the [production URL](/api-reference/application-config#envproductionurl) from your Microsoft Azure project. You can keep the standard Azure URL `https://<project>.1.azurestaticapps.net` or provide your own custom domain.
```
Comment (Rhotimee) [inline website/src/content/deployment-examples/azure.mdx:19]:
```suggestion
Moreover, you need to provide the [production URL](/api-reference/application-config#envproductionurl) from your Microsoft Azure project. You can keep the standard Azure URL `https://<project>.1.azurestaticapps.net` or provide your custom domain.
```

**Review by Rhotimee — APPROVED** [review]
Looks good now. Great job 💪🏽 
I added some minor rephrase suggestions.

File: website/src/content/deployment-examples/azure.mdx [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+* Select your organization, repository and branch.
+* In the **Build details** section select `Custom` as the build preset.
+* Provide the following build settings:
+    * App location: `/`
+    * Output location: `public`
+
+<Info>
+
+If your Custom Application config [requires environment variables](#using-environment-variables), make sure to provide them in your **Home > Static Web Apps > Your project name > Configuration > Application Settings**.
```
Comment (emmenko) [inline website/src/content/deployment-examples/azure.mdx:71]: Is this step needed when using github actions?

File: website/src/content/deployment-examples/azure.mdx [hunk 9]
```diff
@@ -0,0 +1,103 @@
+---
+title: Deploying to Microsoft Azure with Static Web Apps
```
Comment (emmenko) [inline website/src/content/deployment-examples/azure.mdx:2]: I would also rename the file to `azure-static-web-apps`, similarly to `aws-s3-cloudfront`

```suggestion
title: Deploying to Azure with Static Web Apps
```

**Review by emmenko — COMMENTED** [review]
Almost there 🤗

File: website/src/content/deployment-examples/azure.mdx (hunk 8 above)
Comment (kark) [reply website/src/content/deployment-examples/azure.mdx:71]: It is, because the auto-provisioned GH Action is configured to read secrets from the Static Web App env vars config:
https://docs.microsoft.com/en-us/azure/static-web-apps/build-configuration?tabs=github-actions#environment-variables

[This is my example of GH Action](https://github.com/kark/azure-deployment-example-via-portal/blob/95d54f13ac31da0987f9cd7a4981d4736ab67b62/.github/workflows/azure-static-web-apps-red-rock-0ff5a4003.yml#L35) which was created by Azure and those secrets are read from Azure portal config

Comment (emmenko) [reply website/src/content/deployment-examples/azure.mdx:71]: I don't have access to that repo.

Comment (kark) [reply website/src/content/deployment-examples/azure.mdx:71]: Sorry, just made it public

File: website/src/content/deployment-examples/azure.mdx [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+* Provide the following build settings:
+    * App location: `/`
+    * Output location: `public`
+
+<Info>
+
+If your Custom Application config [requires environment variables](#using-environment-variables), make sure to provide them in your **Home > Static Web Apps > Your project name > Configuration > Application Settings**.
+
+Alternatively, when using the [GitHub Action workflow file](https://docs.microsoft.com/en-us/azure/static-web-apps/build-configuration?tabs=github-actions#environment-variables) you need to provide the environment variables there.
+
```
Comment (CarlosCortizasCT) [inline website/src/content/deployment-examples/azure.mdx:74]: I'm a little bit confused here since we say this paragraph is an alternative, but it seems it's mandatory (as it uses the wording **you need to provide**).
I understand you need to first configure your env variables following the instruction above and then you must update your Github flow configuration file.
Is it like that or does it work any other way?

**Review by CarlosCortizasCT — COMMENTED** [review]
Thanks for the new doc and refactoring the others as well to keep it consistent 👏 

I just left a comment to clarify something I didn't fully understand.

File: website/src/content/deployment-examples/azure.mdx (hunk 8 above)
Comment (emmenko) [reply website/src/content/deployment-examples/azure.mdx:71]: So when you connect the repo to Azure, Azure creates the github secrets automatically based on the environment variables that you defined in Azure? And then you just reference them in the github action?

From the example in the docs, it seems to me that you can just provide the variables in the github action, without having to define them in Azure as well.

That's where I'm a bit confused 🤔

Comment (kark) [reply website/src/content/deployment-examples/azure.mdx:71]: My bad, apparently I remembered it all wrong, sorry for all the confusion 😕🤦‍♂️ I remember trying out various options of env vars settings some time ago when I configured that.

Anyhow, you are totally right, the secrets are read from github repo settings and the whole section `If your Custom Application config [requires environment variables](#using-environment-variables), make sure to provide them in your **Home > Static Web Apps > Your project name > Configuration > Application Settings**.` is not needed.

File: website/src/content/deployment-examples/azure.mdx (hunk 10 above)
Comment (kark) [reply website/src/content/deployment-examples/azure.mdx:74]: Thanks, I'm sorry for the confusion. I explained it [here](https://github.com/commercetools/merchant-center-application-kit/pull/2715#discussion_r930804506). That is a good point and I will rewrite that part.

**Comment by kark** [comment]
Thanks everyone for the comments, I updated the docs in bd7005fb880b2e51aa3baff6328bf1e159f10dec

**Review by CarlosCortizasCT — APPROVED** [review]
💯

File: website/src/content/deployment-examples/azure-static-web-apps.mdx [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+* Select your organization, repository and branch.
+* In the **Build details** section select `Custom` as the build preset.
+* Provide the following build settings:
+    * App location: `/`
+    * Output location: `public`
+
+<Info>
+
+If your Custom Application config [requires environment variables](#using-environment-variables), make sure to provide them to your GitHub repository secrets and update the [GitHub Action workflow file](https://docs.microsoft.com/en-us/azure/static-web-apps/build-configuration?tabs=github-actions#environment-variables) accordingly.
```
Comment (emmenko) [inline website/src/content/deployment-examples/azure-static-web-apps.mdx:71]: Nit: I would rephrase this as not all environment variables need to be defined as secrets.

```suggestion
If your Custom Application config [requires environment variables](#using-environment-variables), make sure to provide them in the [GitHub Action workflow file](https://docs.microsoft.com/en-us/azure/static-web-apps/build-configuration?tabs=github-actions#environment-variables). You can define the environment variables either as plain text or using [GitHub encrypted secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets).
```

File: website/src/content/deployment-examples/azure-static-web-apps.mdx [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+
+with:
+  ...
+  action: "upload"
+  app_location: "/"
+  api_location: ""
+  output_location: "public"
+env:
+  APPLICATION_ID: ${{ secrets.APPLICATION_ID }}
+```
```
Comment (emmenko) [inline website/src/content/deployment-examples/azure-static-web-apps.mdx:86]: Let's use a more "complete" example that includes the github action itself

````suggestion
See example below for defining environment variables for the GitHub action:

```yaml title="deployment.yml" highlightLines="9-10"
- name: Build And Deploy
  uses: Azure/static-web-apps-deploy@v1
  with:
    action: "upload"
    app_location: "/"
    api_location: ""
    output_location: "public"
  env:
    APPLICATION_ID: ${{ secrets.APPLICATION_ID }}
    CLOUD_IDENTIFIER: gcp-eu
```
````

**Review by emmenko — COMMENTED** [review]
Thanks, all changes look to me now. Just left a couple of small nitpicks, otherwise 👌

**Review by emmenko — APPROVED** [review]
🚀 Awesome, thanks again for looking into that and documenting the process! 🤗
